packages/editor/src/hooks/custom-sources-backwards-compatibility.js

```diff
@@ -108,7 +107,11 @@ const createWithMetaAttributeSource = ( metaAttributes ) =>
 function shimAttributeSource( settings ) {
 	/** @type {WPMetaAttributeMapping} */
 	const metaAttributes = mapValues(
-		pickBy( settings.attributes, { source: 'meta' } ),
+		Object.fromEntries(
+			Object.entries( settings.attributes ?? {} ).filter(
```

Comment:
`settings.attributes` could theoretically be nullish, so we're falling back to an empty object just in case.